models/nft/ethereum/nft_ethereum_floor_price.sql

```diff
@@ -0,0 +1,126 @@
+{{ config(
+    alias = 'floor_price',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'nft_contract_address'],
+    post_hook='{{ expose_spells(\'["ethereum"]\',
+                                "sector",
+                                "nft",
+                                \'["Henrystats"]\') }}'
+    )
+}}
+
+WITH 
+
+src_data as (
+    SELECT 
+        *,
+        date_trunc('day', block_time) as block_date
+    FROM 
+    {{ ref('nft_trades') }} 
+    WHERE blockchain = 'ethereum'
+    AND trade_type = 'Single Item Trade'
+    AND tx_from != LOWER('0x0000000000000000000000000000000000000000')
+    AND amount_raw > 0 
+    {% if is_incremental() %}
+    AND block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+),
+
+min_trade_date as (
+    SELECT 
+        MIN(block_date) as first_trade_date, 
+        nft_contract_address
+    FROM 
+    src_data
+    GROUP BY 2 
+), 
+
+days as (
+    SELECT 
+        explode(
+            sequence(
+                to_date(first_trade_date), date_trunc('day', now()), interval 1 day -- first trade date in nft.trades
+            )
+        ) as day,
+        nft_contract_address
+    FROM 
+    min_trade_date
+), 
+
+prices as (
+    SELECT 
+        *
+    FROM 
+    {{ source('prices', 'usd') }} prices 
+    WHERE prices.symbol = 'WETH'
+    AND prices.blockchain = 'ethereum'
+    {% if not is_incremental() %}
+    AND prices.minute >= '2017-06-23'
+    {% endif %}
+    {% if is_incremental() %}
+    AND prices.minute >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+), 
+
+prof_data as (
+    SELECT 
+        src.block_date, 
+        src.nft_contract_address,
+        percentile_cont(.05) WITHIN GROUP 
+            (ORDER BY src.amount_original/(
+                CASE 
+                    WHEN src.currency_symbol IN ('ETH', 'WETH') THEN 1 
+                    ELSE prices.price
+                END
```

Comment:
From my understanding this price calculation doesn't hold up.
You're doing:

For ETH/WETH trades
`ETH amount / 1`

For other ERC20 trades
`ERC20 amount / ETH price`

Reply:
fixed